src/service/viewport-impl.js

```diff
@@ -750,6 +750,10 @@ export class ViewportBindingNatural_ {
     if (this.win.document.defaultView) {
       waitForBody(this.win.document, () => {
         this.win.document.body.style.overflow = 'visible';
+        setStyles(this.win.document.body, {
+          overflowX: 'hidden',
+          overflowY: 'auto',
+        });
```

Comment:
Outside of webview is currently not an issue so we want to reduce the surface over which we are trying out changes.

`overflow-y: auto` will create two nested scrolling contexts. There's no way of telling what this would do without very extensive testing. But I'd definitely expect very hard to find/debug bugs. That's why we have `ViewportBindingNaturalIosEmbed_`. If we can work around this - we should try. If not, we might just have to switch to the existing `ViewportBindingNaturalIosEmbed_` where all these problems are addressed.